scripts/cmake/config/default.cmake

```diff
@@ -0,0 +1,144 @@
+# All configurable build properties for Bareflank are defined here, set to 
+# their default values.
+
+# ------------------------------------------------------------------------------
+# Cmake build attributes
+# ------------------------------------------------------------------------------
+
+set(CMAKE_BUILD_TYPE "Release"
+    CACHE STRING
+    "The type of build"
+    )
+set_property(CACHE CMAKE_BUILD_TYPE PROPERTY STRINGS
+    "Release"
+    "Debug"
+    )
+
+# ------------------------------------------------------------------------------
+# Build attributes
+# ------------------------------------------------------------------------------
+
+
+set(BUILD_SHARED_LIBS OFF
+	CACHE BOOL
+    "Build libraries as shared libraries"
+	)
+
+set(BUILD_STATIC_LIBS ON
+	CACHE BOOL
+	"Build libraries as static libraries"
+	)
+
+# ------------------------------------------------------------------------------
+# Developer Features
+# ------------------------------------------------------------------------------
+
+set(ENABLE_UNITTESTING OFF
+	CACHE BOOL
+	"Enable unit testing"
+	)
+
+set(ENABLE_ASAN OFF
```

Comment:
All of these should be in the form:

```
set(ENABLE_ASAN 
    OFF
    CACHE BOOL "Enable clang AddressSanitizer"
)
```

Reply:
Here are my reasons for wanting to use the format I suggested for configurable variables: First, I think that putting the "CACHE <TYPE>" part of the declaration on its own line makes it extremely easy to see the intended type of each configurable option. It is important to get these types right for all variables, because cmake-gui and ccmake use these types to determine how the variables are displayed. Second, by putting the description on its own line, it maximizes the space that the variable's author has to describe what the variable is for (assuming we wrap text at 80 chars?) For example, see TOOLCHAIN_PATH_GSL, which needs the whole line to contain an accurate description. Third, I think that putting the variable's name and default value on the same line looks more intuitive that you're assigning a variable that value. Maybe we can all fight about this on Gitter and come to a community consensus! :P

Reply:
That’s fine, but I’d like to compromise:

```
set(VAR <value>
    CACHE <type>
    “Description”
)
```

Right now, it appears that the indentation in the code (at least what GitHub is showing) is inconsistent, and in the rest of the code, we do not indent the “)”